opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -58,6 +58,8 @@ type NodePool struct {
 	Affinity         *corev1.Affinity            `json:"affinity,omitempty"`
 	Persistence      *PersistenceConfig          `json:"persistence,omitempty"`
 	AdditionalConfig map[string]string           `json:"additionalConfig,omitempty"`
+	Labels           map[string]string           `json:"labels,omitempty"`
+	EnvVars          map[string]string           `json:"envVars,omitempty"`
```

Comment:
Hi @jlamillan 
I am also looking for this feature.
Thanks for making this available. Hopefully, this PR will get in soon.